Make `KeyError` tag (#1398)
python-discord/bot: bot/resources/tags/dict-get.md

> Often while using dictionaries in Python, you may run into `KeyErrors`. This error is raised when you try to access a key that isn't present in your dictionary.\
> While you can use a `try` and `except` block to catch the `KeyError`, Python also gives you some other neat ways to handle them.

Reviewer: ```suggestion
While using dictionaries in Python, you may run into `KeyError`s. This is raised when you try to access a dictionary item that isn't present in the dictionary. Python gives you some neat ways to handle them.
```
This could be condensed and more clear.

Author: I was told to mention the `try - except` option in a single line here, do you think I should omit it?

Reviewer: If you want to do so, include it. I personally think it's a bit redundant.

> While you can use a `try` and `except` block to catch the `KeyError`, Python also gives you some other neat ways to handle them.\

Reviewer 2: This shouldn't contain a backslash since we want to have a line break
```suggestion
While you can use a `try` and `except` block to catch the `KeyError`, Python also gives you some other neat ways to handle them.
```

> __**The `dict.get` method**__

Reviewer: ```suggestion
**Providing default values**
```
We don't really underline sections in the tags. They're generally just bolded, so this would seem more consistent.

The change of wording isn't necessary, but it's something to consider.

> The [`dict.get`](https://docs.python.org/3/library/stdtypes.html#dict.get) method will return the value for the key if it exists, or None (or a default value that you specify) if the key doesn't exist. Hence it will _never raise_ a KeyError.

Reviewer: ```suggestion
The [`dict.get`](https://docs.python.org/3/library/stdtypes.html#dict.get) method will return the value corresponding to the key, if it exists, and None (or a default value that you provide) if the it doesn't. Hence it will _never raise_ a KeyError.
```

Reviewer 3: `...the key, if it exists, and None (or a default value that you provide) if it doesn't...` Small grammar correction :smile_cat:

```py
>>> my_dict = {"foo": 1, "bar": 2}
>>> print(my_dict.get("foobar"))
None

> >>> print(my_dict.get("foobar", 3))    # here 3 is the default value to be returned, in case the key doesn't exist

Reviewer 2: ```suggestion
>>> print(my_dict.get("foobar", 3))    # here 3 is the default value to be returned, because the key doesn't exist
```

> >>> print(my_dict.get("foobar", 3))    # here 3 is the default value to be returned, because the key doesn't exist
> 3

Reviewer: ```suggestion
>>> print(my_dict.get("foobar", 3))  # Here, 3 is specified as the default value
3  # The default value is returned, since the key doesn't exist
```
Let's make sure that the comment follows PEP 8. I'd also like to see this split up in a way like the above, but I'll leave the details to you.

Author: I didn't know how to split the comment whilst also keeping to the REPL look (with the `>>>`), wouldn't a lone comment look a bit odd?

Reviewer: Hmmm. This looks fine to me. It might also be worth considering putting the explanation outside of the code block, if you don't like putting the comment on the new line.

```

> 

Reviewer: ```suggestion
```
This creates a double space when rendered on Discord. Let's remove this space.

> Some other methods that can be used for handling KeyErrors gracefully are the [`dict.setdefault`](https://docs.python.org/3/library/stdtypes.html#dict.setdefault) method, or by using [`collections.defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict) (check out the `!defaultdict` tag).

Reviewer: ```suggestion
Some other methods for handling `KeyError`s gracefully are the [`dict.setdefault`](https://docs.python.org/3/library/stdtypes.html#dict.setdefault) method and [`collections.defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict) (check out the `!defaultdict` tag).
```
Let's make this less verbose.
